### DISCUSSION TITLE
Variant of the job-shop problem

### DISCUSSION THREAD

**Discussion by tungprime** [google/or-tools#3287] [CP-SAT questions] [answered]
Dear Community, 

I have a question regarding a variant of the Job Shop problem (I posted the same question on the mailing list). The job shop problem is discussed here 

https://developers.google.com/optimization/scheduling/job_shop

For the problem that I am working on, all machines (in this case, they are human!) can do the all tasks of a single job. Let's say for simplicity there are two machines and N jobs. Each job has two tasks: task 1 and task 2. As usual, task 1 has to be completed before task 2 can start. I wonder whether you could suggest some references on this topic? If there are codes in or-tools, that would even be better. 

Thank you for your consideration. 

Best wishes, Tung

**Comment by lperron** [comment, marked as answer]
Have you checked the multiple jobshop examples in the code base (jobshop_sat.cc, the different python versions) ?

**Reply by tungprime** [reply]
I am sorry for asking a trivial question but do you mean the following python file (I am more comfortable with python than c++)? 

https://github.com/google/or-tools/blob/stable/examples/python/jobshop_ft06_sat.py

**Comment by lperron** [comment]
Yes or rcpsp_sat.py

Le mar. 10 mai 2022, 20:23, Tung T NGUYEN ***@***.***> a
écrit :
